starlite/middleware/session.py

```diff
@@ -0,0 +1,216 @@
+    def __init__(
+        self,
+        app: "ASGIApp",
+        config: SessionCookieConfig,
+    ):
+        """Starlite SessionMiddleware.
+
+        Args:
+            app: The 'next' ASGI app to call.
+            config: SessionCookieConfig instance.
+        """
+        super().__init__(app)
+        self.app = app
+        self.config = config
+        self.aesgcm = AESGCM(config.secret.get_secret_value())
+
+    def dump_data(self, data: Any) -> List[bytes]:
+        """Given orjson serializable data, including pydantic models and numpy
+        types, dump it into a bytes string, encrypt, encode and split it into
+        chunks of the desirable size.
+
+        Args:
+            data: Data to serialize, encrypt, encode and chunk.
+
+        Notes:
+            - The returned list is composed of a chunks of a single base64 encoded
+                string that is encrypted using AES-CGM.
+
+        Returns:
+            List of encoded bytes string of a maximum length equal to the 'CHUNK_SIZE' constant.
+        """
+        serialized = dumps(data, default=Response.serializer, option=OPT_SERIALIZE_NUMPY)
+        nonce = urandom(NONCE_SIZE)
+        encrypted = self.aesgcm.encrypt(nonce, serialized, associated_data=None)
+        encoded = b64encode(nonce + encrypted)
+        return [encoded[i : i + CHUNK_SIZE] for i in range(0, len(encoded), CHUNK_SIZE)]
+
+    def load_data(self, data: List[bytes]) -> Any:
+        """Given a list of strings, decodes them into the session object.
+
+        Args:
+            data: A list of strings derived from the request's session cookie(s).
+
+        Returns:
+            A deserialized session value.
+        """
+        decoded = b64decode(b"".join(data))
+        nonce = decoded[:NONCE_SIZE]
+        encrypted_session = decoded[NONCE_SIZE:]
+        decrypted = self.aesgcm.decrypt(nonce, encrypted_session, associated_data=None)
+        return loads(decrypted)
+
+    def create_send_wrapper(
+        self, scope: "Scope", send: "Send", should_vacate_session: bool
+    ) -> Callable[["Message"], Awaitable[None]]:
+        """
+        Creates a wrapper for the ASGI send function, which handles setting the cookies on the outgoing response.
+        Args:
+            scope: The ASGI connection scope.
+            send: The ASGI send function.
+            should_vacate_session: A boolean flag dictating whether the session cookie should be vacated on the client
+                side.
+
+        Returns:
+            None.
+        """
+
+        async def wrapped_send(message: "Message") -> None:
+            """A wrapper around the send function, declared in local scope to
+            use closure values.
+
+            Args:
+                message: An ASGI message.
+
+            Returns:
+                None
+            """
+            if message["type"] == "http.response.start":
+                headers = MutableHeaders(scope=message)
+                if should_vacate_session:
```

Comment:
@Goldziher We cannot set empty cookie at client side if the session was corrupted or expired. This is because if the session was expired or corrupted, we should pass an empty session to the user so that the user code can fill it with its own authentication parameters that will be set as a cookie in the response. If the user doesn't make any changes to the session, then in the response, the empty session will be set as a cookie.

I am making changes to it and will show you soon why the design has to be like this.

Reply:
ok, I trust you